lib/BuildSystem/BuildSystem.cpp

```diff
@@ -2180,6 +2183,13 @@ class ShellCommand : public ExternalCommand {
       ctx.error("working-directory unsupported on this platform");
       return false;
 #endif
+    } else if (name == "control-enabled") {
+      if (value != "true" && value != "false") {
+        ctx.error("invalid value: '" + value + "' for attribute '" +
+                  name + "'");
+        return false;
+      }
+      controlEnabled = value == "true";
```

Comment:
Is there a good reason to make this per command? I am concerned about the size impact on the manifest file. Could this be a global setting instead, at least for now while we are still experimenting with it?

Reply:
Hmm, my thought was that a client could selectively disable the control channel for particular commands that are non-conforming.  That said, it was also a minimal resistance path to passing the argument in.  I can definitely see adding a client level default value for it as well, though it will be a little more work to get that value into the externalCommand objects where it is needed.

Reply:
@ddunbar If we do this per-command, some sample stats would be: between 2000-2500 occurrences of this property in a manifest file for a project with around 15k source code files.

Reply:
Hmm, trying to set this at the client level has some annoying plumbing.  But, perhaps a 'tool' level default makes sense.  So the "shell" tool could support this attribute as well and set the default value for all such commands.

Reply:
👍 tool level

Reply:
Tool level sounds perfect